osquery/tables/system/linux/apt_sources.cpp

```diff
@@ -198,6 +210,140 @@ static void genAptSource(const std::string& source,
   }
 }
 
+Status parseDeb822Block(const std::string& input_block,
+                        std::vector<AptSource>& apt_sources) {
+  std::vector<std::string> uris;
+  std::vector<std::string> suites;
+
+  for (const auto& input_line : osquery::split(input_block, "\n")) {
+    // Skip empty lines
+    if (input_line.empty()) {
+      continue;
+    }
+
+    // Remove everything after # from the line for comments
+    auto comment_pos = input_line.find("#");
+    std::string line;
+    if (comment_pos != std::string::npos) {
+      line = input_line.substr(0, comment_pos);
+    } else {
+      line = input_line;
+    }
+
+    auto colon_pos = line.find(":");
+    if (colon_pos == std::string::npos) {
+      continue;
+    }
+
+    auto key = line.substr(0, colon_pos);
+    auto value = line.substr(colon_pos + 1, std::string::npos);
+    boost::trim(value);
+
+    if (value.empty()) {
+      continue;
+    }
+
+    if (key == "Types") {
+      auto types = osquery::split(value);
+      if (std::find(types.begin(), types.end(), "deb") == types.end()) {
+        return Status::failure("not a deb type repo");
+      }
+    }
+
+    if (key == "Enabled" && value != "on") {
+      return Status::failure("repo is not enabled");
```

Comment:
Is it a failure to parse when `Enabled` is `off`? (I would just not return it on the results.)

Reply:
Maybe not, I can make it return a success instead